src/shared/User.ts

```diff
@@ -0,0 +1,7 @@
+export interface User {
+  id?: string | undefined;
+  email?: string | undefined;
+  name?: string | undefined;
```

Comment:
`name?: string | undefined; === name?: string` same applies for above as well

Reply:
Nice catch. You are right, this is subtle but it doesn't work like I expect it to work...
Even with function tsc is still ok with smth like that, while in fact it should not be.
Like if I mark argument as optional, and function has only one argument then what I strictly expect is that either this argument is not passed (`arguments.length = 0`) or it is passed and it is user (`arguments.length = 0 && arguments[0] instance of User`).  But ts actually allows passing undefined as first argument, which is not expected IMHO and should not be allowed. WDYT?


<img width="353" alt="Screenshot 2023-08-07 at 12 34 59" src="https://github.com/kilohealth/web-app-monitoring/assets/133940464/1c9c02be-6e42-4e1e-aad9-78180a57a483">

Reply:
In other words what I am saying that either as func argument or key in the object there is a way to distinguish between passing value as undefined and not passing value. In object we can use smth like `key in object`, in function - smth like arguments length. Though, in function you can avoid passing only last arguments anyway, but you still have a way to check this and can count on that...